src/errors.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use syntax::codemap::FileName;
+
+use std::convert::From;
+use std::io;
+use std::path::PathBuf;
+
+pub type RustfmtResult<T> = Result<T, RustfmtError>;
+
+#[derive(Debug, Fail)]
+pub enum RustfmtError {
```

Comment:
I think we could call these `Error` and `Result` and rely on explicit naming where there is confusion - `RustFmt*` is a bit long. (Just `FmtError` and `FmtResult` would be OK too).